cluster-agent/utils/argocd_login_command.go

```diff
@@ -15,11 +15,18 @@ import (
 // This file is loosely based on the 'argocd login' CLI command (https://github.com/argoproj/argo-cd/blob/0a46d37fc6af9fe0aa963bdd845e3d799aa0320d/cmd/argocd/commands/login.go#L60)
 
 func generateDefaultClientForServerAddress(server string, optionalAuthToken string, skipTLSTest bool) (argocdclient.Client, error) {
+	// managedEnv := &db.ManagedEnvironment{
+	// 	Managedenvironment_id: application.Managed_environment_id,
+	// }
+
+	// clusterCredentials := &db.ClusterCredentials{
+	// 	Clustercredentials_cred_id: managedEnv.Clustercredentials_id,
+	// }
 
 	globalClientOpts := argocdclient.ClientOptions{
 		ConfigPath:           "",
 		ServerAddr:           server,
-		Insecure:             true, // TODO: GITOPSRVCE-178: once support for TLS certification validation is implemented, the value should be used here.
+		Insecure:             true, // clusterCredentials.AllowInsecureSkipTLSVerify TODO: GITOPSRVCE-178: once support for TLS certification validation is implemented, the value should be used here.
```

Comment:
I am doubtful about this, based on my understanding to get `clusterCreds` we need `managedEnv`, for that we need an `application` of type `db.application` and because this function is referenced hierarchically upwards I am not sure whether `application` can be introduced. Do you think this has a workaround?